PYME/config.py

```diff
@@ -508,3 +508,78 @@ def get_init_filename(filename, legacy_scripts_directory=None):
             return fnp
         
     return None
+
+def update_yaml_keys(fn, d):
+    """ 
+    Update the keys in a YAML file without destroying comments.
+
+    TODO: Plays it fast and lose with regex and won't work for many 
+    situations. Currently only works in the case we have a line
+
+    key : value  # optional comment, optional number of spaces
+                 # around the colon, no option for a space at
+                 # the beginning of the line
+    
+    To make this work well, we'll need a full YAML parser 
+    that handles comments. We could either use ruamel.yaml 
+    (https://pypi.org/project/ruamel.yaml/) or write our own.
+
+    Parameters
+    ----------
+    fn : string
+        Path to a yaml file.
+    d : dict
+        key, value pairs of keys/values to update or append to the
+        end of the file
+    """
+    import re
+
+    # Read the yaml file
+    with open(fn) as f:
+        data = f.read()
+
+    # Update the appropriate keys
+    for k, v in d.items():
+        x = re.search('^{}\s*:\s*([^(#|\n|\s)]*)'.format(k),data,flags=re.MULTILINE)
```

Comment:
I thought regex strings needed to either use the `r'my regex'` form or escape the backslashes ... Is this tested?

It also looks as though this will stop on any whitespace (which might not be desirable - it would be good to allow string parameters with whitespace in them). 

I'd try this pattern: `r'^{}\s*:.*$'`.format(k)

Reply:
This is tested in py3. The `r` might be necessary on py2. Your regex will also work, but it might delete a comment on the same line. I think all we'd need to do is remove the `\s` in the final block to not stop on any whitespace. I admittedly only tested with non-whitespace elements after the `:`.

Reply:
I stand corrected. `$` appears to match comments as well (it's the end of the line).

Reply:
Hmm, further testing seems to indicate this search string actually does delete comments on the same line as the variable. Do we care about this?